jenkins/github_helpers.py

```diff
@@ -0,0 +1,180 @@
+"""
+Helper methods for connecting with Github
+"""
+import logging
+import os
+
+from git import Git
+from github import Github, InputGitAuthor, InputGitTreeElement
+
+logging.basicConfig()
+logger = logging.getLogger()
+logger.setLevel(logging.INFO)
+
+
+def get_github_token():
+    """
+    Get the github token environment variable.
+    """
+    try:
+        github_token = os.environ.get('GITHUB_TOKEN')
+    except:
+        raise Exception(
+            "Could not find env variable GITHUB_TOKEN. "
+            "Please make sure the variable is set and try again."
+        )
+    return github_token
+
+
+def get_user_email():
+    """
+    Get the github user email environment variable.
+    """
+    try:
+        github_user_email = os.environ.get('GITHUB_USER_EMAIL')
+    except:
+        raise Exception(
+            "Could not find env variable GITHUB_USER_EMAIL. "
+            "Please make sure the variable is set and try again."
+        )
+    return github_user_email
+
+
+def authenticate_with_github():
+    """
+    Authenticate with Github using a token and return the instance.
+    """
+    github_token = get_github_token()
+    try:
+        github_instance = Github(github_token)
+    except:
+        raise Exception(
+            "Failed connecting to Github. " +
+            "Please make sure the github token is accurate and try again."
+        )
+    return github_instance
+
+
+def connect_to_repo(github_instance, repo_name):
+    """
+    Get the repository object of the desired repo.
+    """
+    repos_list = github_instance.get_user().get_repos()
```

Comment:
It looks like there's an `AuthenticatedUser.get_repo(name)` method, any reason we're not using that here?